include/util/assert.hpp

```diff
@@ -0,0 +1,45 @@
+#ifndef OSRM_UTIL_ASSERT_HPP
+#define OSRM_UTIL_ASSERT_HPP
+
+#include <iostream>
+#include <string>
+
+#include "util/coordinate.hpp"
+
+#include <boost/assert.hpp>
+
+// Enhances BOOST_ASSERT / BOOST_ASSERT_MSG with convenience location printing
+// - OSRM_ASSERT_WITH_LOC(cond, coordinate)
+// - OSRM_ASSERT_WITH_LOC_MSG(cond, coordinate, msg)
+
+#ifdef BOOST_ENABLE_ASSERT_HANDLER
+
+#define OSRM_ASSERT_WITH_LOC_MSG(cond, loc, msg)                                                   \
```

Comment:
Minor nitpick `OSRM_ASSERT` is a little bit shorter to type then `OSRM_ASSERT_WITH_LOC` and since we don't have `OSRM_ASSERT_WITHOUT_LOC` I don't think we need that extra piece of information. :-)